panda_deburring/ocp_croco_force_feedback.py

```diff
@@ -386,199 +406,115 @@ def create_terminal_model(self) -> crocoddyl.ActionModelAbstract:
         terminal_model.differential.armature = self._robot_models.armature
         return terminal_model
 
+    def create_reference_cache(self):
+        models = [self._solver.problem.runningModels]
+        models.append(self._solver.problem.terminalModel)
+
+        for model in models:
+            running_model_diff = model.differential
+
+            # Create new cache object
+            ocp_ref_cache = OCPCosts()
+
+            # By using `__dict__["element"]` obtain references to functions in order to cache them
+            state_reg = running_model_diff.costs.costs["stateReg"]
+            ocp_ref_cache.state_reg = OCPReferences(
+                reference=state_reg.cost.residual.__dict__["reference"],
+                weights=state_reg.cost.activation.__dict__["weights"],
+            )
```

Comment:
I would find it clearer to store the cost instead of the individual items. I don't think it would degrade performances.